pkg/debuginfo/manager.go

```diff
@@ -455,8 +440,8 @@ func (di *Manager) upload(ctx context.Context, dbg *objectfile.ObjectFile) (err
 		size = dbg.Size
 		h    string
 	)
-	if v, ok := di.hashCache.GetIfPresent(key); ok {
-		h = v.(string) //nolint:forcetypeassert
+	if v, ok := di.hashCache.Get(key); ok && v.deadline.After(time.Now()) {
```

Comment:
I feel like we can have an `LRUCacheWithTTL` abstraction to wrap this up. It would be less prone (at least for me 😅 )